src/components/v2/AuthModal/AccountDetails/index.tsx

```diff
@@ -0,0 +1,89 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import Typography from '@mui/material/Typography';
+
+import { Connector } from 'clients/web3';
+import { CHAIN_ID, BscChainId } from 'config';
+import { truncateAddress } from 'utilities/truncateAddress';
+import { Icon } from '../../Icon';
+import { SecondaryButton } from '../../Button';
+import { WALLETS } from '../constants';
+import { useStyles } from './styles';
+
+export interface IAccountDetailsProps {
+  onLogOut: () => void;
+  onCopyAccountAddress: (accountAddress: string) => void;
+  account: {
+    address: string;
+    connector?: Connector;
+  };
+}
+
+export const AccountDetails: React.FC<IAccountDetailsProps> = ({
+  onLogOut,
+  onCopyAccountAddress,
+  account,
+}) => {
+  const styles = useStyles();
+  const truncatedAccountAddress = truncateAddress(account.address);
+
+  // Grab the wallet info. Note that we default to the first wallet in the list
+  // if no match is found, but in reality that case should never happen
+  const { Logo: WalletLogo, name: walletName } =
+    WALLETS.find(wallet => wallet.connector === account.connector) || WALLETS[0];
+
+  const bscScanUrl = `https://${
+    CHAIN_ID === BscChainId.TESTNET ? 'testnet.' : ''
+  }bscscan.com/address/${account.address}`;
+
+  return (
+    <div css={styles.container}>
+      <div css={styles.infoContainer}>
+        <WalletLogo css={styles.walletLogo} />
+
+        <div css={styles.infoRightColumn}>
+          <Typography component="span" css={styles.walletName}>
+            {walletName}
+          </Typography>
+
+          <div css={styles.accountAddressContainer}>
+            <Typography component="span" css={styles.accountAddress}>
+              {account.address}
+            </Typography>
+
+            {/* Only displayed on mobile */}
+            <Typography component="span" css={styles.accountAddressMobile}>
+              {truncatedAccountAddress}
+            </Typography>
+
+            <button
```

Comment:
What is missing that our button component doesn't handle this? This would be "action button", although I've been using the `TextButton` in this situation

Reply:
It is very different from our `button` component. The `button` tag is used semantically here, but UI wise this is what it looks like:
<img width="265" alt="Screen Shot 2022-03-28 at 4 47 10 PM" src="https://user-images.githubusercontent.com/44675210/160436791-5ffe1f9a-220b-4c95-a282-6f21673341a5.png">